File: 1109-corporate-flight-bookings/1109-corporate-flight-bookings.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        output = [0] * n 
        for i in range(len(bookings)):
            start = bookings[i][0]
            end = bookings[i][1]
            seats = bookings[i][2]
            
            output[end - 1] += seats
            
            if start > 1:
                output[start-2] -= seats
        print(output)
        for j in range(n- 2,-1,-1):
            output[j] += output[j+1]
        return output
        
                

File: 1109-corporate-flight-bookings/1109-corporate-flight-bookings.java
class Solution {
    public int[] corpFlightBookings(int[][] bookings, int n) {
        int res[] = new int[n];
        int start;
        int end;
        int seats;
        for(int i = 0; i< bookings.length;i++){
            start = bookings[i][0];
            end = bookings[i][1];
            seats = bookings[i][2];
            
            res[end-1] += seats;
            
            if (start > 1)
                res[start - 2] -= seats;
            
            
        }
        for (int i = n-2; i >= 0; i--){
            res[i] += res[i+1];
        }
        return res;
        
    }
}